checkov/terraform/checks/graph_checks/aws/AWSNATGatewaysshouldbeutilized.yaml

```diff
@@ -0,0 +1,35 @@
+---
+metadata:
+  name: "AWS NAT Gateways should be utilized for the default route"
+  id: "CKV2_AWS_35"
+  guidelines: "This policy identifies Route Tables which have NAT instances for the default route instead of NAT gateways. It is recommended to use NAT gateways as the AWS managed NAT Gateway provides a scalable and resilient method for allowing outbound internet traffic from your private VPC subnets."
+  category: "networking"
+  severity: "medium"
+scope:
+  provider: "aws"
+definition:
+    or:
+        - or:
```

Comment:
@C-Nandhini What's the point of having a nested `or`?

Reply:
We are checking two different resource types route_table and route.
The policy should alert if either one of the resources is incorrectly configured.
Also without the nested OR I got a value error in the editor during testing.
Please advise if there is a different way to achieve the logic.
<img width="818" alt="Screenshot 2022-01-23 at 7 44 48 PM" src="https://user-images.githubusercontent.com/40636098/150682725-ffc6cc78-2bea-479a-b2e2-7996a6b50384.png">

Reply:
@C-Nandhini I do think a flat `or` will achieve the same results, if you want to test it

Reply:
@nimrodkor - Yeah that worked !!!
Thanks for the suggestion, I was of the impression to use OR if the resource type is different.
Surely will correct in future.